neuralprophet/time_net.py

```diff
@@ -513,186 +521,205 @@ def forward_covar_net(self, covariates):
-                while having  ``config_trend.trend_global_local="local"``.
-                The turnaround consists on passing the same meta (dummy ID) to all the samples of the batch.
-                Internally, this is equivalent to use ``config_trend.trend_global_local="global"`` to find the optimal
-                learning rate.
-            compute_components_flag : bool, default=False
-                If True, components will be computed.
+    def forward(
+        self,
+        input_tensor: torch.Tensor,
+        meta: Dict = None,
+        compute_components_flag: bool = False,
+        predict_mode: bool = False,
+    ) -> torch.Tensor:
+        """This method defines the model forward pass."""
 
-        Returns
-        -------
-            torch.Tensor
-                Forecast of dims (batch, n_forecasts, no_quantiles)
-        """
-        # Turnaround to avoid issues when the meta argument is None and meta_used_in_model
+        time_input = unpack_time_feature(
+            input_tensor, self.n_lags, self.n_forecasts, self.max_lags, self.config_model.features_map
+        )
+        # Handle meta argument
         if meta is None and self.meta_used_in_model:
             name_id_dummy = self.id_list[0]
             meta = OrderedDict()
-            meta["df_name"] = [name_id_dummy for _ in range(inputs["time"].shape[0])]
+            meta["df_name"] = [name_id_dummy for _ in range(time_input.shape[0])]
             meta = torch.tensor([self.id_dict[i] for i in meta["df_name"]], device=self.device)
+
+        # Initialize components and nonstationary tensors
         components = {}
         additive_components = torch.zeros(
-            size=(inputs["time"].shape[0], self.n_forecasts, len(self.quantiles)),
+            size=(time_input.shape[0], self.n_forecasts, len(self.quantiles)),
             device=self.device,
         )
         additive_components_nonstationary = torch.zeros(
-            size=(inputs["time"].shape[0], inputs["time"].shape[1], len(self.quantiles)), device=self.device
+            size=(time_input.shape[0], time_input.shape[1], len(self.quantiles)),
+            device=self.device,
         )
         multiplicative_components_nonstationary = torch.zeros(
-            size=(inputs["time"].shape[0], inputs["time"].shape[1], len(self.quantiles)), device=self.device
+            size=(time_input.shape[0], time_input.shape[1], len(self.quantiles)),
+            device=self.device,
         )
-        trend = self.trend(t=inputs["time"], meta=meta)
+
+        # Unpack time feature and compute trend
+        trend = self.trend(t=time_input, meta=meta)
         components["trend"] = trend
 
-        if "seasonalities" in inputs:
-            s = self.seasonality(s=inputs["seasonalities"], meta=meta)
+        # Unpack and process seasonalities
+        seasonalities_input = None
+        if self.config_seasonality and self.config_seasonality.periods:
+            print("++++seasonalities ++++")
```

Comment:
print statement from debugging

Reply:
Still working on cleaning the code 🙈. Let you know once it's ready for review!